scripts/recovery_sync

```diff
@@ -24,6 +24,17 @@ target="$1"
 workdir=$(mktemp -d)
 trap cleanup EXIT
 
+#
+# When syncing the recovery environment, we want to ensure that only a single
+# process is attempting to do this at any given time.  Thus, we add some
```

Comment:
Should probably do this before we create the workdir, as it will end up running mktemp twice, once before the lock, once after.